### DISCUSSION TITLE
Timer parameters dialog is confusing me

### DISCUSSION THREAD

**Discussion by stheine** [arendst/Tasmota#20162] [Q&A]
### PROBLEM DESCRIPTION

The `Configuration / Configure Timer` (Timer parameters) dialog is confusing me:
- There is the radio-button selection for `Time` / `Sunrise` / `Sunset`
- Below, there is the Time picker (Hour and Minute) plus the Offset picker (+/- and Minutes)

If `Time` is selected, only the `Hour` and `Minute` are relevant, right? but why is the offset Minutes selector visible now?
![image](https://github.com/arendst/Tasmota/assets/19614566/a9c699e2-4784-4170-9cc0-2465db2c6545)

And if `Sunrise` or `Sunset` is selected, shouldn't `Hour` and `Minute` be hidden, since only the +/- and Minutes for the Offset be visible?
![image](https://github.com/arendst/Tasmota/assets/19614566/92d69d9a-9391-4b3c-acf0-c2118675396f)

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): `Shelly 1`
- [x] Tasmota binary firmware version number used: `13.2.0`
  - [x] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
07:41:49.188 MQT: tasmota/carport/stat/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
07:41:49.416 MQT: tasmota/carport/stat/RESULT = {"Module":{"46":"Shelly 1"}}
07:41:49.669 MQT: tasmota/carport/stat/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"224":"Relay1"},"GPIO5":{"192":"Switch_n1"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}
```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
07:42:31.029 MQT: tasmota/carport/stat/STATUS = {"Status":{"Module":46,"DeviceName":"Carport","FriendlyName":["Carport"],"Topic":"tasmota","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
07:42:31.035 MQT: tasmota/carport/stat/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:12:03","StartupUTC":"2023-12-04T06:30:28","Sleep":50,"CfgHolder":4617,"BootCount":29,"BCResetTime":"2021-12-03T13:24:08","SaveCount":510,"SaveAddress":"FA000"}}
07:42:31.041 MQT: tasmota/carport/stat/STATUS2 = {"StatusFWR":{"Version":"13.2.0(tasmota)","BuildDateTime":"2023-10-19T09:02:07","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"406/699"}}
07:42:31.046 MQT: tasmota/carport/stat/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["holzhaus",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A002800000000","00000081","00006000","00004000","00000000"]}}
07:42:31.060 MQT: tasmota/carport/stat/STATUS4 = {"StatusMEM":{"ProgramSize":635,"Free":368,"Heap":21,"ProgramFlashSize":1024,"FlashSize":2048,"FlashChipId":"15701C","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62,68","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}}
07:42:31.068 MQT: tasmota/carport/stat/STATUS5 = {"StatusNET":{"Hostname":"tasmota-carport","IPAddress":"192.168.6.31","Gateway":"192.168.6.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.6.6","DNSServer2":"0.0.0.0","Mac":"C4:5B:BE:57:FC:CE","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
07:42:31.074 MQT: tasmota/carport/stat/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.6.7","MqttPort":1883,"MqttClientMask":"Carport","MqttClient":"Carport","MqttUser":"DVES_USER","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
07:42:31.081 MQT: tasmota/carport/stat/STATUS7 = {"StatusTIM":{"UTC":"2023-12-04T06:42:31","Local":"2023-12-04T07:42:31","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":"+01:00","Sunrise":"07:58","Sunset":"16:28"}}
07:42:31.089 MQT: tasmota/carport/stat/STATUS10 = {"StatusSNS":{"Time":"2023-12-04T07:42:31","Switch1":"OFF"}}
07:42:31.095 MQT: tasmota/carport/stat/STATUS11 = {"StatusSTS":{"Time":"2023-12-04T07:42:31","Uptime":"0T00:12:03","UptimeSec":723,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"ON","Wifi":{"AP":1,"SSId":"holzhaus","BSSId":"3C:A6:2F:F1:7A:6C","Channel":6,"Mode":"11n","RSSI":88,"Signal":-56,"LinkCount":1,"Downtime":"0T00:00:03"}}}
```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
see above

### EXPECTED BEHAVIOUR
see above

### SCREENSHOTS
see above

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment]
With Sunrise/Sunset, the hh:mm fields are fixed offsets, you can select `+` or `-` in the extra dropdown appearing to the left.

The `+/-` dropdown to the right is random minutes, up to 15, for a bit of variation.

Note that to ask questions, you should use "Discussions", not "Issues" which is for bugs.

**Reply by stheine** [reply]
so you're saying that in this case, 
![image](https://github.com/arendst/Tasmota/assets/19614566/97223340-b534-48ad-b747-086b9c58c94f)
the timer would trigger at 16:28 + 4:35 +10 => 21:13?

well... ok... with that explanation, I can understand that... but from the configuration dialog I can't.

I'm sorry that I opened this as an issue: I considered this a bug, since I was under the impression that the hiding of the input fields was not properly done.

**Reply by sfromis** [reply]
Not 21:13, but between 20:53 and 21:13, randomly, with your `+/-` selection of 10 minutes.
What you missed was the new dropdown appearing, with a purpose.

Does explains about the fields, in the JSON section where the values map pretty directly to the web UI.
https://tasmota.github.io/docs/Timers/

**Reply by stheine** [reply]
May I propose enhancing the dialog a little bit into something like this:
![image](https://github.com/arendst/Tasmota/assets/19614566/478ccfe9-c72e-4e82-b287-4e98cacc31ce)
that would have helped me a lot to better understand the purpose of the input fields?

**Reply by stheine** [reply]
I created a pull request for the documentation, as that would have helped me to better understand it (it wouldn't really have helped for my UI issues, since I looked at the docs but didn't really find the connection between the UI fields and the documented JSON payload values - but even now, after understanding the UI a little better, I think the docs could be enhanced):
https://github.com/tasmota/docs/pull/1308
